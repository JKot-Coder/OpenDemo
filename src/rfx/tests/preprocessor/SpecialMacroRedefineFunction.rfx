// :RUN_RFXC -P %STD_OUTPUT% %INPUT% 

// 
// Check replacing __FILE__
//

#define __FILE__(x) "Is" x

__FILE__("Anybody")
__FILE__(There)

#undef __FILE__

__FILE__