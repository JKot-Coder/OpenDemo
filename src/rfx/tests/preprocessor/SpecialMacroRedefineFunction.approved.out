SourceOutput:[
    #line 1 "D:/Work/my/OpenDemo/src/rfx/tests/preprocessor/SpecialMacroRedefineFunction.rfx"
    
    
    
    
    
    
    
    
    "Is" "Anybody"
    "Is" There
    
    
    
    __FILE__
]
Diagnostic:[
    D:/Work/my/OpenDemo/src/rfx/tests/preprocessor/SpecialMacroRedefineFunction.rfx:7:9: warning: redefining builtin macro [-Wbuiltin-macro-redefined]
    #define __FILE__(x) "Is" x
            ^
]
